### PR TITLE
chore(upgrade): Prepare major version 2 by removing all the dead code

### PR DESCRIPTION
**Pull request by damienalexandre**
Following the merge of Elastica 8, we drop OpenSearch.

Also it's time to remove the deprecated features.

### CODE REVIEW

File: src/Index.php [hunk 1]
```diff
@@ -29,10 +33,16 @@ public function __construct(Client $client, string $name, ResultSetBuilder $resu
     }
 
     /**
+     * @return mixed|string|null
+     *
+     * @throws ClientResponseException
```
Comment (VincentLanglet) [inline src/Index.php:38]: Instead listing all the exceptions, you might prefer to just add the interface
```
@throws ElasticsearchException
```
in the phpdoc. (Same for others phpdoc where you added `@throws`).

File: src/Index.php [hunk 2]
```diff
@@ -29,10 +33,16 @@ public function __construct(Client $client, string $name, ResultSetBuilder $resu
     }
 
     /**
+     * @return mixed|string|null
```
Comment (VincentLanglet) [inline src/Index.php:36]: This phpdoc seems useleess since there is `mixed` as native return type.

File: src/Index.php (hunk 1 above)
Comment (damienalexandre) [reply src/Index.php:38]: You have a point but when looking at the different exceptions, I see some value in declaring them independently.

We have 4 from elasticsearch/elasticsearch-php:

- ClientResponseException, which has `use ResponseTrait;` which gives hints on the HTTP Response
- ServerResponseException, which has `use ResponseTrait;` which gives hints on the HTTP Response
- MissingParameterException :white_check_mark: ok to use the ElasticsearchException
- NoNodeAvailableException :x: doesn't implement ElasticsearchException

And 2 from other deps:

- ExceptionInterface is from Elastica
- SerializerExceptionInterface is from the Serializer

I think it's nicer to keep them all, so static analysis can warn if someone tries a `$exception->getResponse()` on the wrong one, and still our auto-completion works when developing with it.

That's also used in Elastica itself:

https://github.com/ruflin/Elastica/blob/58042887616eeb63621412c03bc903056bbcee7e/src/Bulk.php#L299-L306

**Comment by damienalexandre** [comment]
Thanks @VincentLanglet for the review :wink:

**Pull request merged**